poetry/core/semver/version.py

```diff
@@ -80,10 +80,24 @@ def is_empty(self) -> bool:
         return False
 
     def allows(self, version: "Version") -> bool:
-        return self == version
+        _this, _other = self, version
+
+        # allow weak equality to allow `3.0.0+local.1` for `3.0.0`
+        if not _this.is_local() and _other.is_local():
+            _other = _other.without_local()
+        elif _this.is_local() and not _other.is_local():
+            _this = _this.without_local()
+
+        # allow weak equality to allow `3.0.0-1` for `3.0.0`
+        if not _this.is_postrelease() and _other.is_postrelease():
+            _other = _other.without_postrelease()
+        elif _this.without_postrelease() and not _other.without_postrelease():
+            _this = _this.without_postrelease()
+
+        return _this == _other
 
     def allows_all(self, other: "VersionTypes") -> bool:
-        return other.is_empty() or other == self
+        return other.is_empty() or other.allows(self)
```

Comment:
We must leave it as is since a single version cannot allow all versions from a range unless this range is the same as the single version.